src/keygen/participant.rs

```diff
@@ -320,10 +343,11 @@ impl KeygenParticipant {
     ) -> Result<ProcessOutcome<<Self as ProtocolParticipant>::Output>> {
         info!("Handling ready keygen message.");
 
-        let (ready_outcome, is_ready) = self.process_ready_message::<storage::Ready>(message)?;
+        let (ready_outcome, is_ready) = self.process_ready_message::<R>(rng, message)?;
```

Comment:
nit: The type of `R` can be inferred here, you don't have to specify it:
```rust
self.process_ready_message(rng, message)?;
```

Ditto at all the call sites for this method.